### PR TITLE
Add color-function-notation

### LINKED ISSUE #4718: Add color-function-notation
> What is the problem you're trying to solve?

I want to disallow the use of (comma-based) legacy color function notation so that I only wrote (space-based) future-facing syntax, and transpile it if necessary with [postcss-preset-env](https://preset-env.cssdb.org/features).

- [Spec](https://www.w3.org/TR/css-color-4/#rgb-functions)
- [CanIUse](https://caniuse.com/#feat=mdn-css_types_color_space_separated_functional_notation)
- [Relevant tweet](https://twitter.com/mathias/status/1253242715304857601)

Legacy notation:

```css
a { color: rgb(0, 0, 0, 0.5) }
```

Modern notation:

```css
a { color: rgb(0 0 0 / 50%) }
```

> What solution would you like to see?

A new rule to do this. Some users, like with pseudo-element colon notation, might want to enforce the legacy style. I propose a rule with two options: "space" and "comma" (unless there are better names?).

Any thoughts on the use case, rule name or options?

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Closes #4718

> Is there anything in the PR that needs further explanation?

I thought I'd give this one a whirl as it's been a while since I wrote a rule.

The `"modern"` option is autofixable, and I've addressed the edge-cases I can think of.

There may be a neater way to write the fixing code that doesn't rely on a count.

### CODE REVIEW

File: lib/rules/color-function-notation/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+<!-- prettier-ignore -->
+```css
+a { color: hsl(.75turn 60% 70%) }
+```
+
+### `"legacy"`
+
+Applicable color-functionss _must always_ use the legacy notation.
```
Comment (m-allanson) [inline lib/rules/color-function-notation/README.md:72]:
```suggestion
Applicable color-functions _must always_ use the legacy notation.
```

Comment (jeddy3) [reply lib/rules/color-function-notation/README.md]: Good catch.

**Comment by jeddy3** [comment]
@malsf21 Thank you for the review! 

> Not the most experienced dev on the topic

With the bugs that you've been squashing, you have more familiarity with stylelint than most! We're a small group of contributors and so any help makes an impact and is appreciated (whether that's squashing bugs, reviewing pull requests or triaging issues!).

---

Let's merge this pull request after we release `13.4.1` (https://github.com/stylelint/stylelint/issues/4791) and when the other two related rules are approved.

**Comment by jeddy3** [comment]
> - Added: `color-function-notation` rule ([#4760](https://github.com/stylelint/stylelint/pull/4760)).

**Pull request merged**